src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -684,6 +714,63 @@ private static bool InitRTNoCacheSupported()
                 "NoCache");
         }
 
+        private void RTServerCertificateCallback(RTHttpBaseProtocolFilter sender, RTHttpServerCustomValidationRequestedEventArgs args)
+        {
+            // Convert WinRT certificate to .NET certificate.
+            X509Certificate2 serverCert = ConvertCertificate(args.ServerCertificate);
+
+            // Create .NET X509Chain from the WinRT information. We need to rebuild
+            // the chain since WinRT only gives us an array of intermediate certificates.
+            var serverChain = new X509Chain();
+            foreach (RTCertificate cert in args.ServerIntermediateCertificates)
+            {
+                serverChain.ChainPolicy.ExtraStore.Add(ConvertCertificate(cert));
+            }
+            serverChain.Build(serverCert);
+
+            // Determine chain and name errors from WinRT information.
+            SslPolicyErrors sslPolicyErrors = SslPolicyErrors.None;
+            if (serverChain.ChainStatus.Length > 0)
+            {
+                sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
+            foreach (RTChainValidationResult result in args.ServerCertificateErrors)
```

Comment:
Wouldn't this be enough to identify the chain errors? If no, is there any reason for not removing / augmenting this check with code we already have for other components (see above comment).